### PR TITLE
Introduction to Network Types and Topologies

### PR DESCRIPTION
**Pull request by kvardhan10**
"Learn different types of networks and different ways of arranging a computer network."

This commit includes one index.md file and 9 pictures(in .jpg format)
This article does not have an idea content **issue**. So no linking here - in the PR.

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tool:** [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you used Repl.it for your code snippets?

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by hectorkambow** [comment]
plagiarism check done ✔️

**Comment by hectorkambow** [comment]
@kvardhan10 Thank you for your articles - reviewing it thoroughly it now - the article is looking great so far - and learning a lot from it. 
I'll be leaving in-line comments on areas where I believe we could add some more value for the audience. 
🚀

File: articles/network-types-topologies/index.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+This enables people to access them from anywhere around the world (securely of course). For this to work, computers are interlinked with one another. This interconnection between two or more computers forms a [computer network](https://en.wikipedia.org/wiki/Computer_network).
+
+They are categorized based on their:
+
+-  Geographical spread of network strength.
+-  Type of usage and applications.
+-  Mode of transmission and some other factors.
+
+A typical networking setup requires two types of devices: **Servers and Workstations.**
+Servers are the computers responsible for powering the network. Workstations are the devices that make up the network. Some of those devices could be printers, laptops, smartphones, etc. Data transfer and resource sharing is done from the servers. The shared materials are downloaded on the workstations in the form of data packets. While some can be used across countries, others can barely be used within a building. Here are some of the most popular types of networks in use.
```
Comment (hectorkambow) [inline articles/network-types-topologies/index.md:28]: This sentence left me wondering as to Why?
"While some can be used across countries, others can barely be used within a building."
Why is it that some packets can travel the world and other can barely travel within buildings? Is it due to the network topology - if so a short sentence clarifying this would help the audience read along.

File: articles/network-types-topologies/index.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+![Example of WLAN](/engineering-education/network-types-topologies/wlan.jpg)
+
+#### Metropolitan Area Network (MAN)
+The point of having [MANs](https://en.wikipedia.org/wiki/Metropolitan_area_network) is similar to LANs. One of the differences is the geographical spread of the network range. As mentioned above, LANs are confined to small areas. But, as the name suggests, MAN’s are spread across an entire metropolitan city. They can be used to share data from one point of the city to another. Business establishments, telephone companies, cable TV’s use this type of networks.
```
Comment (hectorkambow) [inline articles/network-types-topologies/index.md:43]: I think a bit more on MAN would help - as a reader Im left wondering how a MAN is truly different than a LAN - what helps it connect device with such a big range? Is there mini networks in the middle? Is there some hardware involved? 
Any added info to explain how MAN can operate such a big area. 🚀 
is the hardware used more powerful?

File: articles/network-types-topologies/index.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+#### Wide Area Network (WAN)
+This type of network is currently the one mostly used. [WAN](https://en.wikipedia.org/wiki/Wide_area_network) is nothing but a collection of several LANs linked together. These networks are spread across countries and other parts of the world. Some of them are connected through optic fibers while some are connected through satellites. WANs are used by government agencies, multinational companies for exchanging information overseas. Since the network is spread across the world, it is maintained by many firms and companies. When the network is this big, it is prone to failures and system shutdowns once in a while. The biggest WAN in existence is the Internet.   
+
+![Example of WAN](/engineering-education/network-types-topologies/wan.jpg)
+
+ Configuring a computer network can also depend on the extent of the network, the distance between devices and the number of devices used. While LANs, WLANs, and WANs are most commonly used, there are some [other](https://www.belden.com/blog/smart-building/network-types) computer networks that can also be used based on our networking parameters.
```
Comment (hectorkambow) [inline articles/network-types-topologies/index.md:52]: This sentence seems like a transition - but Im not quite sure - is the sentence suggesting that the topologies detailed below are the others that were mentioned? 
If not - a short list of examples of what others? you meant with links would be helpful for any interested reader

File: articles/network-types-topologies/index.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+
+
+### Network Topologies
+Arrangement of components or parts in a specific way is called a **Topology.** In [network topologies](https://en.wikipedia.org/wiki/Network_topology), the components are arranged in a systematic order for *smooth data flow* in a communication network. Here, the components may be devices, workstations, etc which are also referred to as **nodes**. Many topologies are used based on their applications, structure and efficiency. While some are used by commercial establishments and industrial sectors, some are used by schools and for household purposes.
```
Comment (hectorkambow) [inline articles/network-types-topologies/index.md:57]: Can we unpack this sentence:
"Many topologies are used based on their applications, structure and efficiency." 
I think there is some useful info here - but as a beginner - Im not entirely sure what this sentence means.
Perhaps restructuring the sentence would help. 
If it makes sense we can also omit the sentence- if the meaning and or message is handled somewhere else.

File: articles/network-types-topologies/index.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+#### Linear Topology
```
Comment (hectorkambow) [inline articles/network-types-topologies/index.md:59]: Nice section - It may be worth noting this type of topology is outdated and/or any advantages and disadvantages that may come with using this type of topology. 🚀

File: articles/network-types-topologies/index.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+![Linear Topology](/engineering-education/network-types-topologies/bus.jpg)
+
+#### Ring Topology
+Ring topology is a special kind of bus topology. *A linear topology in which both ends are joined together forms a **Ring Topology***. Thus no terminators are used and so the data flow is *unidirectional*. Because of that, for the data to reach from the first node to the last  it has to traverse through all the nodes. Since the connection of nodes is continuous, data flows infinitely throughout the topology until one node removes it.
+Like linear topology, the structure and arrangement of nodes are very simple. Due to its compact architecture, the configuration of nodes becomes easier. Since the nodes are connected in a loop, unpredictable troubleshooting problems may appear. To repair or fix one crashed node, the whole system has to be taken down. It is difficult to shut down one node and keep the rest of the system functional. Like the linear topology, adding new devices and components may slow down the data transfer rate.
```
Comment (hectorkambow) [inline articles/network-types-topologies/index.md:67]: Same as the comment above - I think you've already stated some advantages and disadvantages but making it clear to the reader for help with flow and readability. 
Also is this network type used much today?

File: articles/network-types-topologies/index.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+![Ring Topology](/engineering-education/network-types-topologies/ring.jpg)
+
+#### Mesh Topology
+Mesh topology is the solution to the problem with ring topology. Unlike ring topology, here *every node is connected with every other node present in the network*. In the case of a single node failure, the rest of the system can still be functional. Since one node connects many other nodes, data is shared through any one of the many paths available. This makes mesh arrangement a suitable network for industries and business establishments. A mesh topology in which all the nodes are connected to each other is called full connectivity, and the mesh topology in which only some nodes are connected to each other is called partial connectivity.
```
Comment (hectorkambow) [inline articles/network-types-topologies/index.md:72]: very nice section as well - I think maybe rearranging some sentences in an advantage/disadvantage format - doesn't have to be a list - the sentence can simply state:
Mesh topology can come with many benefits such as....
"Since this increases power consumption to a larger extent...it can be seen as a disadvantage to the Mesh topology."
"Some organizations may see the "proper planning and requires high maintenance" as a disadvantage. 
Something around these line - feel free to modify any way you see fit.

We just want the reader to be able to clearly understand the facts stated .

File: articles/network-types-topologies/index.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+Mesh topology increases the network flow and data can be shared faster. It also records a lesser number of network breakages and system breakdowns. But since all the nodes are connected to each other, each node low key acts as a router to another node and so on. This increases power consumption to a larger extent. Mesh topology needs lots of proper planning and requires high maintenance. Due to many links, high network latency may be an issue.
+
+![Mesh Topology](/engineering-education/network-types-topologies/mesh.jpg)
+
+#### Star Topology
+*This topology consists of an extra part, i.e. hub or switch, in the center of the topology which connects all the nodes*. A hub or switch is a networking device that transfers data from one device to another device. It acts as a transmission medium. While both are used depending on their operations, some [differences](https://www.geeksforgeeks.org/difference-between-hub-and-switch/) may affect the way they perform. This is the most commonly used topology.
```
Comment (hectorkambow) [inline articles/network-types-topologies/index.md:78]: "This is the most commonly used topology." 
Maybe a link supporting this claim and/or an explanation as to why its the most common - what makes it the ideal choice? 🚀

File: articles/network-types-topologies/index.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+
+![Star Topology](/engineering-education/network-types-topologies/star.jpg)
+
+#### Tree Topology
+Tree topology is a special kind of bus topology. Here *central nodes or central hubs are connected to the main bus*. These nodes or hubs are also called as **root nodes**. The other peripheral nodes are connected to those nodes or hubs. These peripheral nodes are called as **descendant nodes**. Like the bus topology, terminators are attached at both ends to define the length of the main bus. This type of topology is used where the data has to be shared in a hierarchical manner.
+If one branch or one descendant node has problems it will not affect the entire network. Adding more branches to the main bus becomes easier. But the main bus connects many sub-branches and if it has any complications, then it affects the whole network. This may also lead to network shut down. Thus, it becomes hard to arrange and group the nodes. Also, it costs very much to install this type of topology.
+
+![Tree Topology](/engineering-education/network-types-topologies/tree.jpg)
+
```
Comment (hectorkambow) [inline articles/network-types-topologies/index.md:88]: A short conclusion would wrap this up nicely - maybe sharing your thoughts on what is the best topology based on what we learned. 
Or perhaps what you believe the next level (of normal) will be for topologies - what innovation or advancement do you for see.

File: articles/network-types-topologies/index.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+In an emergency of network failure, identifying the damaged node becomes easier. This is because a single hub connects all the nodes. Here, the damaged node will be shut down and the rest of the nodes can still be functional. But if a hub is damaged then all the nodes connected to that hub will be shut down for reconstruction.
+
+![Star Topology](/engineering-education/network-types-topologies/star.jpg)
+
+#### Tree Topology
+Tree topology is a special kind of bus topology. Here *central nodes or central hubs are connected to the main bus*. These nodes or hubs are also called as **root nodes**. The other peripheral nodes are connected to those nodes or hubs. These peripheral nodes are called as **descendant nodes**. Like the bus topology, terminators are attached at both ends to define the length of the main bus. This type of topology is used where the data has to be shared in a hierarchical manner.
```
Comment (hectorkambow) [inline articles/network-types-topologies/index.md:84]: "shared in a hierarchical manner." 
Perhaps an example

**Comment by hectorkambow** [comment]
@kvardhan10  Im assuming this PR was approved before we implemented the Idea Content Form in GitHub? 🤔

**Comment by kvardhan10** [comment]
> @kvardhan10 Im assuming this PR was approved before we implemented the Idea Content Form in GitHub?

Yes! On 14th September. Idea Content Form in GitHub was implemented on 17th.

File: articles/network-types-topologies/index.md (hunk 1 above)
Comment (kvardhan10) [reply articles/network-types-topologies/index.md]: Right. So I've placed a new topic on network transmission and how they affect the range/extent of the network.
I would like to provide an external link to the "differences between bridges and repeaters". But I feel that the readers should be familiar with the OSI model and whatabouts of data link layer and physical layer. What's your call on this @hectorkambow ?

File: articles/network-types-topologies/index.md (hunk 2 above)
Comment (kvardhan10) [reply articles/network-types-topologies/index.md]: I've included the transmission methods and the devices used in a MAN. Also provided a link to Modems.

File: articles/network-types-topologies/index.md (hunk 3 above)
Comment (kvardhan10) [reply articles/network-types-topologies/index.md]: Well, a topology is like a design/model in which network types can be structured. The details mentioned above have nothing to do with the topologies. The above section is like a conclusion to the "different types of networks".

Yes - by "others"(which is a reference link) I meant the other network types like Personal Area Network(PAN- used by an indiviual) or Campus Area Network(CAN- used by a campus) or even a Virtual Private Network(VPN- used to create a virtual p2p connection to provide privacy to the users.)

Is the single link enough or shall I mention a couple of them(by giving separate links) and then the "others(reference link)"?

File: articles/network-types-topologies/index.md (hunk 4 above)
Comment (kvardhan10) [reply articles/network-types-topologies/index.md]: Yes- the types of topologies based on their structure, applications are explained in their respective parts. I feel that the sentence is redundant in the intro part, so I'll consider removing it.

File: articles/network-types-topologies/index.md (hunk 5 above)
Comment (kvardhan10) [reply articles/network-types-topologies/index.md:57]: Added two new points. Made a separate list of advantages and disadvantages.
Also added a new point regarding its less usage in the communication system.

File: articles/network-types-topologies/index.md (hunk 6 above)
Comment (kvardhan10) [reply articles/network-types-topologies/index.md]: Sure- I'll do one thing. I'll make 2 separate parts(Advantages and Disadvantages) for each topology and list the points there.

Well, kind of. The data sent by the source computer passes through all the computers to reach the destination computer. This allows all the computers to "see" what the data is. This helps when data has to be shared publicly with more people. Like in our school, teachers used to share these lecture slides and PPTs from their computer(source: teacher) to all the computers simultaneously and they would start appearing one by one in our computer(destination: student) as they traversed throughout the system. Our teachers said that all the computers were connected in a ring manner. But nowadays, star topology is considered to be a leading structure/model to form a communication network. Since a central hub is used, the administrator controlling the hub decides what data to send to which computer, which is kind of risk-free and reliable.
So basically, ring topology is not suited for commercial and business purposes. But it's ok to use when the data security is not an issue or the transmission is between trusted computers. Also, ring is cheaper and requires less cabling than compared to star.

File: articles/network-types-topologies/index.md (hunk 7 above)
Comment (kvardhan10) [reply articles/network-types-topologies/index.md]: Restructured the paragraph. I felt the advantage/disadvantage(A/D) format as a list is more helpful and clearer to differentiate between the points.

File: articles/network-types-topologies/index.md (hunk 8 above)
Comment (kvardhan10) [reply articles/network-types-topologies/index.md]: Restructured the paragraph(A/D).
Added another point as to why star topology is the most used one.

File: articles/network-types-topologies/index.md (hunk 10 above)
Comment (kvardhan10) [reply articles/network-types-topologies/index.md]: Provided examples

File: articles/network-types-topologies/index.md (hunk 9 above)
Comment (kvardhan10) [reply articles/network-types-topologies/index.md:138]: Restructured the paragraph(A/D).
Added conclusion part.

**Comment by kvardhan10** [comment]
Updated the index.md file.
Details are in the above in-line comments.

**Comment by kvardhan10** [comment]
> @kvardhan10 Im assuming this PR was approved before we implemented the Idea Content Form in GitHub?

Yes! On 14th September. The new Idea Content Form in GitHub was implemented on the 17th.
